### PR TITLE
*: Add support for `Or` expressions

### LINKED ISSUE #146: Add `Or` expression
While we already have the `And` expression, we cannot express `Or` expressions today.

### PR DESCRIPTION
**Pull request by MadhavJivrajani**
This change adds support for `Or` expressions, make changes to:
- `logicalplan`
- `physicalplan`

Also modifies the example to demonstrate it.

TODO:
- [x] Add test coverage

Fixes #146

### CODE REVIEW

File: query/logicalplan/expr.go [hunk 1]
```diff
@@ -243,6 +246,39 @@ func and(exprs []Expr) Expr {
 	}
 }
 
+func Or(exprs ...Expr) Expr {
+	return or(exprs)
+}
+
+func or(exprs []Expr) Expr {
```
Comment (MadhavJivrajani) [inline query/logicalplan/expr.go:253]: We could probably also reuse most of this code from the `and` recursive function

File: filter.go [hunk 2]
```diff
@@ -114,6 +129,29 @@ func (a *AndExpr) Eval(rg dynparquet.DynamicRowGroup) (bool, error) {
...
+		return true, nil
+	}
+
+	right, err := a.Right.Eval(rg)
+	if err != nil {
+		return false, err
+	}
+
+	// This stores the result in place to avoid allocations.
+	return left || right, nil
```
[the remark this answers is not in the document]
Comment (MadhavJivrajani) [reply filter.go:152]: ooo good point, thanks!

File: filter.go [hunk 3]
```diff
@@ -114,6 +129,29 @@ func (a *AndExpr) Eval(rg dynparquet.DynamicRowGroup) (bool, error) {
...
+	if left {
+		return true, nil
+	}
+
+	right, err := a.Right.Eval(rg)
+	if err != nil {
+		return false, err
+	}
+
+	// This stores the result in place to avoid allocations.
```
Comment (brancz) [inline filter.go:151]: This comment can be removed now, it’s misleading.

**Comment by brancz** [comment]
Looks amazing! Thank you so much for the awesome contribution!

**Pull request merged**